src/tesseract.pybind.h

```diff
@@ -243,33 +245,75 @@ void add_tesseract_module(py::module& root) {
         config : TesseractConfig
             The configuration object for the decoder.
       )pbdoc")
-      .def("decode_to_errors",
-           py::overload_cast<const std::vector<uint64_t>&>(&TesseractDecoder::decode_to_errors),
-           py::arg("detections"),
-           py::call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>(), R"pbdoc(
+      .def(
+          "decode_to_errors",
+          [](TesseractDecoder& self, const py::array_t<bool>& syndrome) {
+            if ((size_t)syndrome.size() != self.num_detectors) {
+              std::ostringstream msg;
+              msg << "Syndrome array size (" << syndrome.size()
+                  << ") does not match the number of detectors in the decoder ("
+                  << self.num_detectors << ").";
+              throw std::invalid_argument(msg.str());
```

Comment:
you don't need the stringstream 
```suggestion
             std::string msg = "Syndrome array size (" + std:to_string(syndrome.size())
                  + ") does not match the number of detectors in the decoder ("
                   +  std::to_string(self.num_detectors) + ")."
              throw std::invalid_argument(msg);
```